encoder/vaapiencoder_h264.cpp

```diff
@@ -1275,6 +1299,82 @@ YamiStatus VaapiEncoderH264::getCodecConfig(VideoEncOutputBuffer* outBuffer)
     return m_headers->getCodecConfig(outBuffer);
 }
 
+void VaapiEncoderH264::fill(
+    VAEncMiscParameterTemporalLayerStructure* layerParam) const
+{
+    layerParam->number_of_layers = m_temporalLayerNum;
+    layerParam->periodicity = H264_MIN_TEMPORAL_GOP;
+
+    for (uint32_t i = 0; i < layerParam->periodicity; i++)
+        layerParam->layer_id[i] = getTemporalId(m_temporalLayerNum, i + 1);
+}
+
+void VaapiEncoderH264::fill(VAEncMiscParameterRateControl* rateControl,
+                            uint32_t temporalId) const
```

Comment:
Instead of duplicate code here, could we just add bitRate param to base class's like:
void VaapiEncoderBase:fill(VAEncMiscParameterRateControl* rateControl,  uint32_t bitRate) const